modules/3d/src/rgbd/odometry_functions.cpp

```diff
@@ -48,13 +48,18 @@ static UMat prepareScaledDepth(OdometryFrame& frame)
     frame.getDepth(depth);
     CV_Assert(!depth.empty());
 
+    UMat depthFlt;
+    depth.convertTo(depthFlt, CV_32FC1);
```

Comment:
Why do you need conversion here? Why not just patchNans directly?

Reply:
Because the input type can also be 8U, 16U, 32S...